extension/src/plots/model/collect.test.ts

```diff
@@ -48,9 +48,10 @@ describe('collectCustomPlots', () => {
     expect(data).toStrictEqual(expectedOutput)
   })
 
-  it('should return only custom plots if there no selected revisions', () => {
-    const expectedOutput: CustomPlotData[] = customPlotsFixture.plots.filter(
-      plot => plot.type !== CustomPlotType.CHECKPOINT
+  it('should return checkpoint plots with empty values if there no selected revisions', () => {
+    const expectedOutput: CustomPlotData[] = customPlotsFixture.plots.map(
+      plot =>
+        plot.type === CustomPlotType.CHECKPOINT ? { ...plot, values: [] } : plot
```

Comment:
This feels odd: 

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/37993418/228112493-b2a2f8ad-1b46-439c-b71b-eb7afc5d4718.png">

I don't know

1. Which plot is which
2. What type of plot should be inside (trends or metrics vs params).

If there are no experiments at all then we end up with this:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/37993418/228112598-b5dd0e91-8499-42de-968f-b8fc12ffd60f.png">

Seems inconsistent.

At the very least I think the text inside of the empty box needs to mention the plot title/select a checkpoint experiment to display but I think they should be hidden with some message to say "select a checkpoint experiment to display trend plots"

Note: Chances of users seeing these screens are minimal due to the use of checkpoint experiments but I think we should attempt to get it right.

Reply:
Also, need to consider when there are no custom plots added but we do have data available:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/37993418/228119461-486a287c-b3f5-4337-b7e6-fed4c782cd1a.png">


An update will definitely be needed after https://github.com/iterative/vscode-dvc/pull/3567. Probably want to add a button.

Reply:
> At the very least I think the text inside of the empty box needs to mention the plot title/select a checkpoint experiment to display but I think they should be hidden with some message to say "select a checkpoint experiment to display trend plots"

A message makes sense! Not sure where it could go though 🤔. Will start experimenting with some ideas :) 

> Also, need to consider when there are no custom plots added but we do have data available:

What's the issue with the current setup? Text says "No Plots Added" and you can add a plot with the plus button 🤔

Reply:
> What's the issue with the current setup? Text says "No Plots Added" and you can add a plot with the plus button

The screen shouldn't be split. The custom section should be hidden and there should be two buttons, one to add an experiment and one to add a custom plot.

Reply:
There are 3 things that can currently cause an "empty" plots screen:

1. No plots in the project
2. No plots selected
3. No experiments selected

I'm about to introduce a global error too which gives 4.

I think we need to distinguish between the user having added custom plots or not.

From those we end up with the following combinations:

| project has plots | plots selected | experiments selected | project has custom plots | project has global error | welcome screen content | buttons to add/create | show or hide custom plots section | 
| -- | -- | -- | -- | -- | -- | -- | -- | 
| N/A | N/A | N/A | N | Y |  Error message | custom plots | hide | 
| N | N/A | N/A | N | N | Info on adding plots to the project | custom plot | hide |
| Y | N | N | N | N | There are no selected plots or experiments | plots, experiments, custom plots | hide |
| Y | N | Y | N | N | There are no selected plots or custom plots | plots, custom plots | hide |
| Y | Y | N | N | N | The are no selected experiments or custom plots | experiments, custom plots | hide | 
| N/A | N/A | N/A | Y | Y | Error message | custom plots  | show |
| N | N/A | N/A | Y | N | Info on adding plots to the project | none | show |
| Y | N | N | Y | N | There are no selected plots or experiments | plots, experiments | show |
| Y | N | Y | Y | N | There are no selected plots | plots  | show |
| Y | Y | N | Y | N | There are no experiments selected | experiments  | show |
| Y | Y | Y | Y | N | Normal webview | N/A | show |

I added suggested welcome screen content, buttons and whether I think we should show or hide the custom plots section. IMO the section should be hidden if there is no other data sent the webview and a button should be added to create new custom plots.

I even think we should do this when there is a global CLI error.

The alternative is that we throw away the welcome screen concept in favour of improving the empty state of the existing sections. We then only have to deal with the project having no data at all.

Reply:
Another alternative is to simplify the welcome screen into as few states as possible.

| project has plots | plots selected | experiments selected | project has custom plots | project has global error | welcome screen content | buttons to select/create | show or hide custom plots section | 
| -- | -- | -- | -- | -- | -- | -- | -- | 
| N | N/A | N/A | N | N | Info on adding plots to the project | custom plot | hide |
| Y | N/A | N/A | N | N | There are no plots to show | experiments, plots, custom plots | hide | 
| N | N/A | N/A | Y | N | Info on adding plots to the project | none | show |
| Y | N/A | N/A | Y | N | There are no plots to show | experiments, plots, experiments | show |

We can deal with global errors separately but that would give us 6 states instead of 11. Seems pretty reasonable to me.

Reply:
Thanks for the clarification! Updated the logic.